torch/_dynamo/variables/lazy.py

```diff
@@ -0,0 +1,108 @@
+import functools
+from typing import Optional
+
+from .base import MutableLocalBase, MutableLocalSource, VariableTracker
+
+
+class LazyMutableLocal(MutableLocalBase):
+    """Container to cache the real VariableTracker"""
+
+    def __init__(self):
+        super().__init__(MutableLocalSource.Local)
+        self.vt: Optional[VariableTracker] = None
+
+
+class LazyVariableTracker(VariableTracker):
+    """
+    A structure that defers the creation of the actual VariableTracker
+    for a given underlying value until it is accessed.
+
+    The `realize` function invokes VariableBuilder to produce the real object.
+    Once a LazyVariableTracker has been realized, internal bookkeeping will
+    prevent double realization.
+
+    This object should be utilized for processing containers, or objects that
+    reference other objects where we may not want to take on creating all the
+    VariableTrackers right away.
+    """
+
+    _nonvar_fields = ["_value"]
+
+    def __init__(self, _value, source, **kwargs):
+        super().__init__(source=source, **kwargs)
+        self._value = _value
+        if self.mutable_local is None:
+            self.mutable_local = LazyMutableLocal()
+        # TODO(voz) - Support SourcelessBuilder during realize
```

Comment:
No point in doing this, since SourcelessBuilder can't create guards

Reply:
Fair point. I was thinking there might be value for LazyVariableTracker for intermediary types, but, thinking more, its not that useful.